src/components/Filter.tsx

```diff
@@ -1,21 +1,37 @@
-import React, { useEffect, useCallback, useRef } from 'react'
+import React, { FC, ReactNode, useCallback, useEffect, useRef } from 'react'
 import { AmiableForm, useForm, useSubmit } from 'amiable-forms'
 import { useManifest } from 'use-manifest'
 
 import { fromUrl as sortsFromUrl } from '../utils/sortsConverter'
 import useUrlParamState from '../hooks/useUrlParamState'
 
-const NOPE = () => false
+const NOPE = (): boolean => false
 
-const Updater = ({ urlState, defaultValues, defaultTableState }) => {
+interface UpdaterProps {
+  urlState: {
+    [key: string]: any
+  }
+  defaultValues?: {
+    [key: string]: any
+  }
+  defaultTableState: {
+    [key: string]: any
+  }
+}
+
+interface UpdaterFilter {
+  [key: string]: any
+}
+
+const Updater = ({ urlState, defaultValues, defaultTableState }: UpdaterProps): null => {
   const { updateState: updateManifestState } = useManifest()
   const { setValues: updateForm } = useForm({ shouldUpdate: NOPE })
-  const filterRef = useRef()
+  const filterRef = useRef<UpdaterFilter>()
 
   useEffect(() => {
     const updatedFilter = {
-      ...(defaultValues || {}),
-      ...(defaultTableState || {}),
+      ...(defaultValues ?? {}),
```

Comment:
good change

Reply:
Plus it was this or have typescript point a gun (shaped a lot like the word 'any') at me